Feature/dcc 5709 icgc python bootstrap project

KevinFHartmann opened the pull request.

KevinFHartmann committed:
Enhancement: Initial checkin of icgc module with setup script

KevinFHartmann committed:
Enhancement: Added LICENSE.txt to make setup.py happy.

KevinFHartmann committed:
Bugfix: The GPL Version classifier has to match the wording on PyPi

KevinFHartmann committed:
Bugfix: Name should be icgc, not icgc-client, to match the module name

andricDu reviewed (changes requested):
Good stuff in here.

I recomment running pylint or something similar on the code however to ensure that you follow the PEP8 style guide.

KevinFHartmann committed:
Enhancement: Code now returns the data from the
Enhancement: The url to connect to defaults to "http://dcc.icgc.org"
Enhancement: The authentication defaults to None (currently unused)
Enhancement: The client now returns JSON objects, not a HttpResponse object.
Code Cleanup: Code now passes all Pycharm code inspection and pylint tests.

KevinFHartmann committed:
Bugfix: Forgot to include copyright headers for the python files. Fixed.
